toucan_connectors/google_big_query/google_big_query_connector.py

```diff
@@ -27,8 +27,14 @@ class GoogleBigQueryDataSource(ToucanDataSource):
     )
 
 
-# TODO - check bad return type
-def _define_type(value) -> str:
+BigQueryParam = Union[bigquery.ScalarQueryParameter, bigquery.ArrayQueryParameter]
+
+
+# NOTE: This does not play nicely with dates. They're a bit tricky
+# though, as we'd have to try and parse dates from strings to
+# determine if something is a date or not. Until then, we can just
+# use a cast. eg: SELECT * FROM table WHERE STRING(date_col) IN UNNEST({{my_dates}})
+def _define_scalar_type(value: Any) -> str:
```

Comment:
I dug a bit into the bigquery lib and found https://github.com/googleapis/python-bigquery/blob/v2.34.3/google/cloud/bigquery/dbapi/_helpers.py#L68 which seem to do the same job.
I went to see that because I was intrigued when I read [this doc](https://googleapis.dev/python/bigquery/2.34.3/dbapi.html#providing-explicit-type-information) saying that using the DB-API, the lib would determine the parameters type itself.

Maybe that's worth a shot to simplify our connector?

Reply:
Nice, thanks :+1: Is it fine with you if I merge this as is, and make another refactor PR to use these functions rather than ours ?